api/src/test/java/org/openmrs/api/VisitServiceTest.java

```diff
@@ -938,5 +938,24 @@ public void getAllVisitTypes_shouldGetAllVisitTypesBasedOnIncludeRetiredFlag() {
 		visitTypes = visitService.getAllVisitTypes(false);
 		assertEquals("get all visit types excluding retired", 2, visitTypes.size());
 	}
+
+	/**
+	 * @see VisitService#purgeVisit(Visit)
+	 */
+	@Test
+	public void purgeVisit_shouldPurgeVisit() {
```

Comment:
the test you wrote already exists at method `purgeVisit_shouldEraseTheVisitFromTheDatabase` in this test class.

two tests of method `VisitService.purgeVisit` are missing, you can see [coveralls for the missing branches that need tests](https://coveralls.io/builds/14803127/source?filename=api%2Fsrc%2Fmain%2Fjava%2Forg%2Fopenmrs%2Fapi%2Fimpl%2FVisitServiceImpl.java)

you should add a separate test for when one would try to purge
* a visit which has not yet been saved: like `VisitService.purgeVisit(new Visit())` (check that the originalSize is unchanged)
* a visit has active encounters (so it is considered to be in use): `Context.getEncounterService().getEncountersByVisit(visit, true).isEmpty()` is `false`

Reply:
@teleivo Thanks, I will restart work on this issue